openmp/libomptarget/src/device.cpp

```diff
@@ -319,14 +319,21 @@ void *DeviceTy::getOrAllocTgtPtr(void *HstPtrBegin, void *HstPtrBase,
        DPxPTR(HstPtrBegin), Size, DPxPTR(lr.Entry->HstPtrBegin),
        lr.Entry->HstPtrEnd - lr.Entry->HstPtrBegin);
     // Explicit extension of mapped data - not allowed.
-    if (HasPresentModifier || !HasNoAllocModifier)
+    int openacc_error_flag = 1;
+    if ((HasPresentModifier && openacc_error_flag == 0) || (!HasNoAllocModifier && openacc_error_flag == 0))
```

Comment:
I think we would need a mechnism where we do not need to set the flag on each possible `if` that issue a diagnostic but more hidden in the `MESSAGE` macro (which can be replaced maybe).